src/Features/Core/Portable/ConvertTupleToStruct/AbstractConvertTupleToStructCodeRefactoringProvider.cs

```diff
@@ -143,35 +144,15 @@ private static string GetTitle(Scope scope)
         private async Task<(SyntaxNode, INamedTypeSymbol)> TryGetTupleInfoAsync(
             Document document, TextSpan span, CancellationToken cancellationToken)
         {
-            var position = span.Start;
-            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var token = root.FindToken(position);
-
-            // Span actually has to be within the token (i.e. not in trivia around it).
-            if (!token.Span.IntersectsWith(position))
-            {
-                return default;
-            }
-
-            if (!span.IsEmpty && span != token.Span)
+            // Enable refactoring either for TupleExpression or TupleType
+            var expressionOrType = await document.TryGetRelevantNodeAsync<TTupleExpressionSyntax>(span, cancellationToken).ConfigureAwait(false) as SyntaxNode;
```

Comment:
consider:

```
var expressionOrType =
    await document.TryGetRelevantNodeAsync<TTupleExpressionSyntax>(span, cancellationToken).ConfigureAwait(false) ??
    await document.TryGetRelevantNodeAsync<TTupleTypeSyntax>(span, cancellationToken).ConfigureAwait(false);
```

Reply:
also, consider an extension method for TryGetRelevantNodeAsync that takes multiple type params.

Reply:
I'm not sure about multiple params, it might not be as obvious and due to possibility to chain ?? doesn't really add much value.

Reply:
fine with me :)